xdsl/dialects/gpu.py

```diff
@@ -240,6 +241,17 @@ def get(dim: DimensionAttr) -> GridDimOp:
                                attributes={"dimension": dim})
 
 
+@irdl_op_definition
+class HostRegisterOp(Operation):
```

Comment:
At this point I am starting to get towards the limits of my technical GPU knowledge. I am guessing that this hosts a memref to the GPU? 

Maybe this is a good point to point out that many of this operations could use some documentation (the others can be done in a different PR) :)

Reply:
It is at my limits too!
This makes the Memref accessible from the device/GPU, and still accessible but the host.
How exactly is deliberately not clear I think, GPU being a high-level dialect to abstract vendors and specific models.

Basically I couldn't document that more than what MLIR did. Should I copy/rephrase it here in docstring?

Reply:
Yeah, I think there is value in having their docstring here instead of having to look it up.

Reply:
Sure, makes sense!

Reply:
@webmiche docstring added, do you have a comment on it?
It's copy-pasted from MLIR's doc directly, so I wouldn't change it too much (as we use their "semantics" ATM), but rephrasing could be fine.

Reply:
LGTM!